gapii/cc/spy.cpp

```diff
@@ -536,22 +578,24 @@ void Spy::onPostFrameBoundary(bool isStartOfFrame) {
     mEncoder->object(&timestamp);
   }
 
-  if (!is_suspended() && mCaptureFrames >= 1) {
-    mCaptureFrames -= 1;
-    if (mCaptureFrames == 0) {
-      mEncoder->flush();
-      mConnection->close();
-      set_suspended(true);
+  if (is_suspended()) {
+    if (mSuspendCaptureFrames > 0) {
+      if (--mSuspendCaptureFrames == 0) {
+        GAPID_DEBUG("Started capture");
+        // We must change suspended state BEFORE releasing the Spy lock with
+        // exit(), because the suspended state affects concurrent CallObservers.
+        set_suspended(false);
+        exit();
+        saveInitialState();
+        enter("RecreateState", 2);
+      }
     }
-  }
-  if (mSuspendCaptureFrames.load() > 0) {
-    if (is_suspended() && mSuspendCaptureFrames.fetch_sub(1) == 1) {
-      // We must change suspended state BEFORE releasing the Spy lock with
-      // exit(), because the suspended state affects concurrent CallObservers.
-      set_suspended(false);
-      exit();
-      saveInitialState();
-      enter("RecreateState", 2);
+  } else {
+    if (mCaptureFrames > 0) {
+      if (--mCaptureFrames == 0) {
+        mCaptureFrames = -1;
```

Comment:
And here `if (mCaptureFrames <= 0)` instead of the two lines

Reply:
No, the second line includes a pre-decrement, that's why it should be separate.